File: InventoryManager.Angular/inventory-manager-frontend/src/app/models/item.ts
import { EntityBase } from "./entity-base";

export class Item extends EntityBase {

  id: string;
  name: string;
  schemaId: string;
  quantity: number;
  properties;

  public constructor(init?: Partial<Item>) {
    super();
    Object.assign(this, init);
  }
}

File: InventoryManager.Api/Models/Item.cs
using System.Collections.Generic;

namespace InventoryManager.Api.Models
{
    public class Item : EntityBase
    {
        public string Name { get; set; }
        public string Type { get; set; }
        public int Quantity { get; set; }
        public IDictionary<string, object> Properties { get; set; }
    }
}